File: example/kotlin/src/main/kotlin/com/some/pkg/somemodule/context/Web.kt
package com.some.pkg.somemodule.context

import com.github.bratek20.architecture.context.api.ContextBuilder
import com.github.bratek20.architecture.context.api.ContextModule
import com.github.bratek20.infrastructure.httpserver.api.WebServerModule

import com.some.pkg.somemodule.api.*
import com.some.pkg.somemodule.web.*

class SomeModuleWebClient(
    private val serverUrl: String = "SOME_SERVER_URL"
): ContextModule {
    override fun apply(builder: ContextBuilder) {
        builder
            .setImplObject(SomeModuleWebServerUrl::class.java, SomeModuleWebServerUrl(serverUrl))
            .setImpl(SomeInterface::class.java, SomeInterfaceWebClient::class.java)
    }
}

class SomeModuleWebServer: WebServerModule {
    override fun getImpl(): ContextModule {
        return SomeModuleImpl()
    }

    override fun getControllers(): List<Class<*>> {
        return listOf(
            SomeInterfaceController::class.java
        )
    }
}

File: example/c-sharp/SomeModule/Context/Web.cs
namespace SomeModule.Context
{
    public class SomeModuleWebClient : ContextModule
    {
        private readonly HttpClientConfig config;

        public SomeModuleWebClient(HttpClientConfig config)
        {
            this.config = config;
        }

        public void Apply(ContextBuilder builder)
        {
            builder
                .SetImplObject<SomeModuleWebClientConfig>(new SomeModuleWebClientConfig(_config))
                .SetImpl<SomeInterface, SomeInterfaceWebClient>()
                .SetImpl<SomeInterface2, SomeInterface2WebClient>();
        }
    }
}